progressReport/app.py

```diff
@@ -125,18 +131,62 @@ def assign_node_levels(node, student_levels_count, class_levels_count):
                            course_node_count=course_node_count,
                            course_data=course_nodes)
 
+def parse_json_mastery(school_name, course_name, student_mastery_json, class_mastery_json):
+    def assign_node_levels_json(node, student_levels_count, class_levels_count):
+        nonlocal student_mastery, class_mastery
+        if not node["children"]:
```

Comment:
Revise this to remove side effects and just return the student and class mastery